src/common/crypto/platform/openssl/crypto.cpp

```diff
@@ -306,75 +305,18 @@ ExpectedPrivateKey PrivateKey::Generate(const unsigned int bits, const unsigned
 			"Failed to generate a private key. Initialization failed: "
 				+ GetOpenSSLErrorMessage()));
 	}
-
-	ret = EVP_PKEY_CTX_set_rsa_keygen_bits(pkey_gen_ctx.get(), bits);
-	if (ret != OPENSSL_SUCCESS) {
-		return expected::unexpected(MakeError(
-			SetupError,
-			"Failed to generate a private key. Parameters setting failed: "
-				+ GetOpenSSLErrorMessage()));
-	}
-
-	auto exponent_bn = unique_ptr<BIGNUM, void (*)(BIGNUM *)>(BN_new(), bn_free);
-	ret = BN_set_word(exponent_bn.get(), exponent);
-	if (ret != OPENSSL_SUCCESS) {
-		return expected::unexpected(MakeError(
-			SetupError,
-			"Failed to generate a private key. Parameters setting failed: "
-				+ GetOpenSSLErrorMessage()));
-	}
-
-	ret = EVP_PKEY_CTX_set_rsa_keygen_pubexp(pkey_gen_ctx.get(), exponent_bn.get());
-	if (ret != OPENSSL_SUCCESS) {
-		return expected::unexpected(MakeError(
-			SetupError,
-			"Failed to generate a private key. Parameters setting failed: "
-				+ GetOpenSSLErrorMessage()));
-	}
-	exponent_bn.release();
-
 	EVP_PKEY *pkey = nullptr;
+#ifdef MENDER_CRYPTO_OPENSSL_LEGACY
+	// https://docs.openssl.org/master/man3/EVP_PKEY_keygen
```

Comment:
What exactly are we referring to from the manual?

Reply:
If I remember correctly it was that keygen does the same thing as generate, I guess I should elaborate further in the comment